api/main.go

```diff
@@ -693,13 +883,18 @@ func (serv *MetadataServer) checkProviderHealth(ctx context.Context, providerNam
 	return statusErr
 }
 
-func (serv *MetadataServer) CreateSourceVariant(ctx context.Context, source *pb.SourceVariant) (*pb.Empty, error) {
-	serv.Logger.Infow("Creating Source Variant", "name", source.Name, "variant", source.Variant)
+func (serv *MetadataServer) CreateSourceVariant(ctx context.Context, sourceRequest *pb.SourceVariantRequest) (*pb.Empty, error) {
+	requestID, ctx, logger := serv.Logger.InitializeRequestID(ctx)
+	logger = logger.WithResource("source", sourceRequest.SourceVariant.Name, sourceRequest.SourceVariant.Variant).WithProvider(logging.SkipProviderType, sourceRequest.SourceVariant.Provider)
```

Comment:
super nit: if you want, for better readability you can format this like, since we're making this a bit "functional" and chaining methods we can also probably format it like that:

```suggestion
	logger = logger.
		WithResource("source",
			sourceRequest.SourceVariant.Name,
			sourceRequest.SourceVariant.Variant,
		).
		WithProvider(
			logging.SkipProviderType,
			sourceRequest.SourceVariant.Provider,
		)
```